lib/frame_window_filtering/hog_labeling.h

```diff
@@ -0,0 +1,54 @@
+/// \file
+/// Maintainer: Felice Serena
+///
+
+#pragma once
+
+#include "frame_window_filtering.h"
+
+#include "classifier/classifier.h"
+
+#include <memory>
+
+namespace MouseTrack {
+
+class HogLabeling : public FrameWindowFiltering {
+public:
+  typedef Classifier::Mat Mat;
+  typedef Classifier::Vec Vec;
+  void train(const Mat &X_train, const Vec &y_train);
+  virtual FrameWindow operator()(const FrameWindow &window) const;
+
+private:
+  std::unique_ptr<Classifier> _classifier;
+  // highest label + 1: we need to know, how many labels there are
+  int _numLabels;
+
+  /// width of the sliding window
+  int _windowWidth = 64;
```

Comment:
Shouldn't we be able to set all these parameters? For example, 64 is quite an arbitrary value we chose

Reply:
Agreed, but I'm not sure what/how to expose them.
I'm a bit afraid of just exposing everything, as I have no idea for constraints between parameters (example: does window width need to be a multiple of cell width?).

I think, for a first iteration, I'll add the "necessary" parameters like "windowWidth/Heigh" and "stepSize" such that we can play around with them.